### PR TITLE
Change output and baseline roots for anvil and chrysalis

### PR DESCRIPTION
**Pull request by jgfouca**

### CODE REVIEW

**Comment by jgfouca** [comment]
@rljacob

**Review by rljacob — APPROVED** [review]

**Comment by amametjanov** [comment]
@jgfouca please make baseline-dirs group-writeable with something like:
```
$ chmod -R g+w /lcrc/group/e3sm2/baselines/chrys/intel/master/
$ chmod -R g+w /lcrc/group/e3sm2/baselines/anvil/intel/master/
or 
$ chmod -R g+w /lcrc/group/e3sm2/baselines/
```

**Comment by jgfouca** [comment]
@amametjanov , at first glance, the permissions look correct:

```
% ls -ld /lcrc/group/e3sm2/baselines/anvil/intel/master/
drwxrwxr-x 53 ac.jgfouca E3SM 16384 Oct 16 19:06 /lcrc/group/e3sm2/baselines/anvil/intel/master/
```

Group-writable and under the `E3SM` metagroup. Are you in that group?

**Comment by rljacob** [comment]
I think ACLs are being used here.  I'll add Az to the E3SM2 group which should enable writing.

File: util/chrysalis_setup.sh [hunk 1]
```diff
@@ -4,3 +4,5 @@ source  /lcrc/soft/climate/e3sm-unified/load_latest_cime_env.sh
 source /gpfs/fs1/soft/chrysalis/spack/opt/spack/linux-centos8-x86_64/gcc-9.3.0/lmod-8.3-5be73rg/lmod/lmod/init/sh
 
 export PROJECT=e3sm
+
+RUNSCRIPT_FLAGS="--scratch-root /lcrc/group/e3sm2/`whoami`/scratch/chrys --baseline-root /lcrc/group/e3sm2/baselines/chrys/intel"
```
Comment (amametjanov) [inline util/chrysalis_setup.sh:8]: @jgfouca @rljacob is it okay to remove this flag?
```
--baseline-root /lcrc/group/e3sm2/baselines/chrys/intel
```
The new path has been sym-linked from old path:
```
$ ll /lcrc/group/e3sm/baselines
lrwxrwxrwx 1 ac.jgfouca E3SM 27 Nov  3  2023 /lcrc/group/e3sm/baselines -> /lcrc/group/e3sm2/baselines
```
When tests are launched with a compiler other than `intel` (hard-wired here), CIME compares to default baseline root at `/lcrc/group/e3sm/baselines/anvil/$COMPILER` and comparisons fail: e.g. `e3sm_landice_developer` path
```
/lcrc/group/e3sm/baselines/chrys/intel/master/SMS.ne30_oECv3_gis.IGELM_MLI.chrysalis_gnu.elm-extrasnowlayers/
```

Comment (rljacob) [reply util/chrysalis_setup.sh:8]: Yes that can be removed.

Comment (amametjanov) [reply util/chrysalis_setup.sh:8]: Done in https://github.com/E3SM-Project/E3SM_test_scripts/commit/a1d47fbf697369884fc138af8c4ef57dfdc2c202 .
Also, I moved baselines for 4 tests in `e3sm_landice_developer`+`gnu` from `intel` to `gnu` path here
```
/lcrc/group/e3sm/baselines/chrys/gnu/master/
```

**Pull request merged**